File: XString48.cpp
#include "pt4.h"
using namespace std;

void Solve()
{
    Task("XString48");
    string s;
    pt >> s;
    s = s + ' ';
    int m;
    int p;
    m = s.find_first_not_of(" ");
    p = s.find(" ");
    for (int i = m+1; i <= p - 1; i++)
    {
        if (s[i] == s[m] && i != m)
        {
            s[i] = '.';
        }
        if (i == p - 1)
        {
            m = s.find_first_not_of(" ", p+1);
            p = s.find(" ", m);
        }
    }
    int j = s.rfind(" ");
    s[j] = '\0';
    pt << s;
}

File: XString48.cs
// File: "XString48"
using PT4;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace PT4Tasks
{
    public class MyTask : PT
    {
        public static async void Solve()
        {
            Task("XString48");//
            string s = GetString();
            int m;
            int p;
            s = s + " ";
            m = 0;
            while (true)
            {
                if (s[m].Equals(' '))
                {
                    m++;
                }
                else
                    break;
            }
            p = s.IndexOf(" ");
            for (int i = m; i <= p - 1; i++)
            {
                if (s[i] == s[m] && i != m)
                {
                    s = s.Remove(i, 1);
                    s = s.Insert(i, ".");
                }
                if (i == p - 1)
                {
                    m = s.IndexOf(" ", p + 1);
                    if(m == -1){
                        break;
                    }
                    while (true)
                    {
                        if (s[m].Equals(' '))
                        {
                            m++;
                        }
                        else
                            break;
                    }
                    i = m;
                    p = s.IndexOf(" ", m);
                }
            }
            s = s.Remove(s.Length-1 , 1);
            Put(s);
        }
    }
}
